core/src/main.rs

```diff
@@ -122,33 +119,21 @@ async fn main() -> Result<(), CoordinatorError> {
         configure_default_log_capture()
     }
 
-    // Construct a starknet client that workers will use to communicate with
-    // Starknet
-    let starknet_client = StarknetClient::new(StarknetClientConfig {
+    // Construct an arbitrum client that workers will use for submitting txs
+    let arbitrum_client = ArbitrumClient::new(ArbitrumClientConfig {
+        darkpool_addr: args.contract_address.clone(),
         chain: args.chain_id,
-        contract_addr: args.contract_address.clone(),
-        infura_api_key: None,
-        starknet_json_rpc_addr: args.starknet_jsonrpc_node.clone().unwrap(),
-        starknet_account_addresses: args.starknet_account_addresses,
-        starknet_pkeys: args.starknet_private_keys,
-    });
+        rpc_url: args.rpc_url.unwrap(),
+        arb_priv_key: args.arbitrum_private_key.clone(),
+    })
+    .await
+    .map_err(|e| CoordinatorError::Arbitrum(e.to_string()))?;
 
     // Build a task driver that may be used to spawn long-lived asynchronous tasks
     // that are common among workers
     let task_driver_config = TaskDriverConfig::default_with_bus(system_bus.clone());
     let task_driver = TaskDriver::new(task_driver_config);
 
-    // Spawn a thread to sync the relayer-global state with on-chain state and
-    // network state
-    let task = InitializeStateTask::new(
```

Comment:
do we no longer need this? and if not, should we delete the struct definition as well?

Reply:
correct this was used for passing in wallets directly via the config, we now just use the API. I removed the task definition in #285